Epos sat

davidcordeiro-deimos opened the pull request:
Fixes #

### Features:
New harvester for the EPOS-Sat product. This harvester allows to collect five new collections:

- Unwrapped Interferogram ("inu")
- Wrapped Interferogram  ("inw")
- LOS Displacement Timeseries ("dts")
- Spatial Coherence ("coh")
- Interferogram APS Global Model ("aps")
- Map of LOS Vector ("cosneu")

Depending on the configuration, a different collection will be collected (**collection code is presented above between brackets**). So, 5 different harvesters must be created with the configurations below:

{
  "collection": "inw",
  "start_date": "2010-01-16T10:30:00Z",
  "timeout": 4,
  "make_private":  false
}

The current number of products is low, due to the fact that currently there are only sample data. A large quantity of data is expected to start being injected in September of 2019. **Thus, you only have to run each harvester once.**

- [X] includes new  features
- [ ] includes tests covering changes
- [X] includes updated documentation
- [ ] includes user-visible changes
- [ ] includes bugfix

Please [X] all the boxes above that apply

davidcordeiro-deimos commented:
Currently waiting for the data provider to provide the official logo, brief description of the project, as well as collection descriptions

davidcordeiro-deimos commented:
The build is failing for a reason unrelated to this PR, it is failing in a Sentinel Harvester test.

joaandrade commented:
**Project (data provider) description:**

The European Plate Observing System (EPOS) is a Pan-European Research Infrastructure (RI) devoted to the integration of existing (and future) advanced European facilities into a single, distributed, sustainable infrastructure for solid Earth Science in Europe. EPOS is articulated in several Thematic Core Services (TCS), which represent the different Solid Earth communities, that generate data, products, software and services accessible and discoverable throught the Integrated Core Service (ICS). Among several, the Satellite Data TCS aims at implementing satellite EO services transverse to the EPOS community and suitable to be used in several solid Earth science application scenarios.
The EO products shared within NextGEOSS are generated by the Satellite Data TCS community of EPOS. At the moment they consist of Earth surface deformation maps (and related products) derived from Copernicus SAR data.

Project logo:
https://www.epos-ip.org/sites/all/themes/divine/logo.png

davidcordeiro-deimos commented:
@JGulic 
The bug has been fixed by preemptively declaring the variable. Although the variable 'collection' should be always set due to the restrictions in harvester configuration.

If you find any other problem please let me know.

JGulic commented:
@davidcordeiro-deimos now the harvester finishes, without an error, but there are no datasets shown on the dataset page. 
This is the error from the logs:

`2019-05-23 13:42:47,832 ERROR [ckan.model.modification] Expecting ',' delimiter: line 1 column 47 (char 46)
Traceback (most recent call last):
  File "/home/gulic/ckan/lib/nextgeoss/src/ckan/ckan/model/modification.py", line 88, in notify
    observer.notify(entity, operation)
  File "/home/gulic/ckan/lib/nextgeoss/src/ckan/ckan/lib/search/__init__.py", line 123, in notify
    operation
  File "/home/gulic/ckan/lib/nextgeoss/src/ckan/ckan/lib/search/__init__.py", line 95, in dispatch_by_operation
    index.insert_dict(entity)
  File "/home/gulic/ckan/lib/nextgeoss/src/ckan/ckan/lib/search/index.py", line 76, in insert_dict
    return self.update_dict(data)
  File "/home/gulic/ckan/lib/nextgeoss/src/ckan/ckan/lib/search/index.py", line 101, in update_dict
    self.index_package(pkg_dict, defer_commit)
  File "/home/gulic/ckan/lib/nextgeoss/src/ckan/ckan/lib/search/index.py", line 278, in index_package
    pkg_dict = item.before_index(pkg_dict)
  File "/home/gulic/ckan/lib/nextgeoss/src/ckanext-nextgeoss/ckanext/nextgeoss/plugin.py", line 229, in before_index
    geometry = json.loads(geometry)
  File "/usr/lib64/python2.7/json/__init__.py", line 339, in loads
    return _default_decoder.decode(s)
  File "/usr/lib64/python2.7/json/decoder.py", line 364, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib64/python2.7/json/decoder.py", line 380, in raw_decode
    obj, end = self.scan_once(s, idx)
ValueError: Expecting ',' delimiter: line 1 column 47 (char 46)
2019-05-23 13:42:47,897 DEBUG [ckanext.nextgeossharvest.lib.nextgeoss_base.import] Package c8222f3a-ef3d-4358-8992-1661f615ec01 was successully harvested.
`

Note: Take a look at the format of the spatial information.

davidcordeiro-deimos commented:
Fixed the spatial field, there was commas missing between latitude and longitude coordinates.

If you find any other issue please let me know.

georgiana-b commented:
@davidcordeiro-deimos I just tested this again. The harvester seems to collect the files but then at fails at indexing and remains hanged. Here is the indexing error:

```
ckan-dev_1    | 2019-05-28 12:07:13,818 ERROR [ckanext.spatial.plugin] Geometry not valid GeoJSON, not indexing
ckan-dev_1    | 2019-05-28 12:07:13,891 ERROR [ckan.lib.search] Error while indexing dataset ddb01785-601e-4534-8d45-cb21a0e0b9e1: ValueError("Expecting ',' delimiter: line 1 column 49 (char 48)",)
ckan-dev_1    | Traceback (most recent call last):
ckan-dev_1    |   File "/usr/bin/paster", line 11, in <module>
ckan-dev_1    |     load_entry_point('PasteScript==2.0.2', 'console_scripts', 'paster')()
ckan-dev_1    |   File "/usr/lib/python2.7/site-packages/paste/script/command.py", line 102, in run
ckan-dev_1    |     invoke(command, command_name, options, args[1:])
ckan-dev_1    |   File "/usr/lib/python2.7/site-packages/paste/script/command.py", line 141, in invoke
ckan-dev_1    |     exit_code = runner.run(args)
ckan-dev_1    |   File "/usr/lib/python2.7/site-packages/paste/script/command.py", line 236, in run
ckan-dev_1    |     result = self.command()
ckan-dev_1    |   File "/srv/app/src/ckan/ckan/lib/cli.py", line 554, in command
ckan-dev_1    |     self.rebuild()
ckan-dev_1    |   File "/srv/app/src/ckan/ckan/lib/cli.py", line 577, in rebuild
ckan-dev_1    |     quiet=self.options.quiet)
ckan-dev_1    |   File "/srv/app/src/ckan/ckan/lib/search/__init__.py", line 194, in rebuild
ckan-dev_1    |     defer_commit
ckan-dev_1    |   File "/srv/app/src/ckan/ckan/lib/search/index.py", line 101, in update_dict
ckan-dev_1    |     self.index_package(pkg_dict, defer_commit)
ckan-dev_1    |   File "/srv/app/src/ckan/ckan/lib/search/index.py", line 278, in index_package
ckan-dev_1    |     pkg_dict = item.before_index(pkg_dict)
ckan-dev_1    |   File "/srv/app/src_extensions/ckanext-nextgeoss/ckanext/nextgeoss/plugin.py", line 231, in before_index
ckan-dev_1    |     geometry = json.loads(geometry)
ckan-dev_1    |   File "/usr/lib/python2.7/json/__init__.py", line 339, in loads
ckan-dev_1    |     return _default_decoder.decode(s)
ckan-dev_1    |   File "/usr/lib/python2.7/json/decoder.py", line 364, in decode
ckan-dev_1    |     obj, end = self.raw_decode(s, idx=_w(s, 0).end())
ckan-dev_1    |   File "/usr/lib/python2.7/json/decoder.py", line 380, in raw_decode
ckan-dev_1    |     obj, end = self.scan_once(s, idx)
ckan-dev_1    | ValueError: Expecting ',' delimiter: line 1 column 49 (char 48)
```

davidcordeiro-deimos commented:
@georgiana-b This error seems to be the same as the one that Jovanka mentioned in a previous comment. I believe that I corrected this issue in the last commit.

Could you please check if indeed you have the most recent version of the code? If so, please let me know, and I look into it.

georgiana-b commented:
@davidcordeiro-deimos I made sure I run the latest version of the code. I started a new harvester and left it running over night. I got that error again and it still didn't finish.

Are you saying you don't get this error at all?

davidcordeiro-deimos commented:
@georgiana-b The error that you showed in a previous comment seems to be related to the 'spatial' field not having the correct format. So if possible could you print and send a 'spatial' field of one dataset that you are obtaining?

Also due to the fact that currently there are only sample data, you do not need to let it run during the night. The collection collection with the most datasets only has 9 datasets, thus the harvest is quite fast.

Please let me know what you find.

georgiana-b commented:
Dear David, I understand that it shouldn't run for so long. That is one of the problems I'm trying to point out:  if it only has 9 datasets it should stop after it gets them. The fact that it's been running the whole night is a bug with the harvester.

I will check what the content of the spatial field is. But you still didn't answer whether you encounter that indexing error or not. If we are to debug this we need to know if we get the same result.

georgiana-b commented:
@davidcordeiro-deimos Here is one of the datasets that failed:

```xml
<entry>
	<id>
https://catalog.terradue.com:443/gep-epos/search?format=atom&uid=InW_GDM-CNRS_20170621_20170627_0001
</id>
	<title type="text">InW_GDM-CNRS_20170621_20170627_0001</title>
	<summary type="html">
		<table>
			<tbody>
				<tr>
					<td>
						<a href="https://sandbox.zenodo.org/record/246071/files/GDM_20170621-20170627_sd_era_4rlks.int.png" target="_blank" title="View preview image">
							<img align="left" border="0" width="100px" title="View preview image" src="https://sandbox.zenodo.org/record/246071/files/GDM_20170621-20170627_sd_era_4rlks.int.png">
							</a>
						</td>
					</tr>
					<tr>
						<td>Product identifier</td>
						<td>
							<strong>InW_GDM-CNRS_20170621_20170627_0001</strong>
						</td>
					</tr>
					<tr>
						<td>Platform</td>
						<td>
							<strong>S1</strong>
						</td>
					</tr>
					<tr>
						<td>Sensor Type</td>
						<td>
							<strong>RADAR</strong>
						</td>
					</tr>
					<tr>
						<td>Product Type</td>
						<td>
							<strong>WRAPPED_INTERFEROGRAM</strong>
						</td>
					</tr>
					<tr>
						<td>Processing center</td>
						<td>
							<strong>GDM-CNRS</strong>
						</td>
					</tr>
					<tr>
						<td>Date of measurement start</td>
						<td>
							<strong>2017-06-21T18:09:22Z</strong>
						</td>
					</tr>
					<tr>
						<td>Date of measurement end</td>
						<td>
							<strong>2017-06-27T18:10:05Z</strong>
						</td>
					</tr>
					<tr>
						<td>Processor name</td>
						<td>
							<strong>WP12-DDSS-024 GDM-SAR Sentinel-1 processing on demand</strong>
						</td>
					</tr>
					<tr>
						<td>Orbit direction</td>
						<td>
							<strong>ASCENDING</strong>
						</td>
					</tr>
					<tr>
						<td>Orbit</td>
						<td>
							<strong>001</strong>
						</td>
					</tr>
					<tr>
						<td>Unit of measure</td>
						<td>
							<strong>rad</strong>
						</td>
					</tr>
					<tr>
						<td>Reference system</td>
						<td>
							<strong>RADAR_GEOMETRY</strong>
						</td>
					</tr>
					<tr>
						<td>License</td>
						<td>
							<strong>https://creativecommons.org/licenses/by-sa/4.0/</strong>
						</td>
					</tr>
					<tr>
						<td>Super Master SAR Image</td>
						<td>
							<strong>S1B_IW_SLC__1SDV_20170703T180938_20170703T181005_006327_00B1F9_012B.SAFE</strong>
						</td>
					</tr>
					<tr>
						<td>Master SAR Image</td>
						<td>
							<strong>S1B_IW_SLC__1SDV_20170621T180937_20170621T181004_006152_00ACE9_35A7.SAFE</strong>
						</td>
					</tr>
					<tr>
						<td>Slave</td>
						<td>
							<strong>S1A_IW_SLC__1SDV_20170627T181009_20170627T181036_017223_01CBB8_3A0F.SAFE</strong>
						</td>
					</tr>
					<tr>
						<td>Perpendicular baseline (m)</td>
						<td>
							<strong>-63.2066448083</strong>
						</td>
					</tr>
					<tr>
						<td>DEM</td>
						<td>
							<strong></strong>
						</td>
					</tr>
					<tr>
						<td>Method</td>
						<td>
							<strong>NSBAS processing chain, differential interferogram generation with bursts coregistration using spectral diversity</strong>
						</td>
					</tr>
					<tr>
						<td>Applied filter</td>
						<td>
							<strong></strong>
						</td>
					</tr>
				</tbody>
			</table>
		</summary>
		<published>2019-01-18T14:51:52.783465Z</published>
		<updated>2019-01-18T14:51:53.119166Z</updated>
		<link href="https://sandbox.zenodo.org/record/246071/files/GDM_20170621-20170627_sd_era_4rlks.int.tiff" rel="enclosure" title="Product URL" type="text/html"/>
		<link href="https://sandbox.zenodo.org/record/246071/files/GDM_20170621-20170627_sd_era_4rlks.int.png" rel="icon" title="Preview URL" type="image/x-icon"/>
		<link href="https://catalog.terradue.com//eop/gep-epos/profile/xml?uid=InW_GDM-CNRS_20170621_20170627_0001" rel="alternate" title="EOP profile" type="application/xml"/>
		<link href="https://catalog.terradue.com/gep-epos/search?format=atom&uid=InW_GDM-CNRS_20170621_20170627_0001" rel="self" title="Reference link" type="application/atom+xml"/>
		<content type="html">
			<table>
				<tbody>
					<tr>
						<td>
							<a href="https://sandbox.zenodo.org/record/246071/files/GDM_20170621-20170627_sd_era_4rlks.int.png" target="_blank" title="View preview image">
								<img align="left" border="0" width="100px" title="View preview image" src="https://sandbox.zenodo.org/record/246071/files/GDM_20170621-20170627_sd_era_4rlks.int.png">
								</a>
							</td>
						</tr>
						<tr>
							<td>Product identifier</td>
							<td>
								<strong>InW_GDM-CNRS_20170621_20170627_0001</strong>
							</td>
						</tr>
						<tr>
							<td>Platform</td>
							<td>
								<strong>S1</strong>
							</td>
						</tr>
						<tr>
							<td>Sensor Type</td>
							<td>
								<strong>RADAR</strong>
							</td>
						</tr>
						<tr>
							<td>Product Type</td>
							<td>
								<strong>WRAPPED_INTERFEROGRAM</strong>
							</td>
						</tr>
						<tr>
							<td>Processing center</td>
							<td>
								<strong>GDM-CNRS</strong>
							</td>
						</tr>
						<tr>
							<td>Date of measurement start</td>
							<td>
								<strong>2017-06-21T18:09:22Z</strong>
							</td>
						</tr>
						<tr>
							<td>Date of measurement end</td>
							<td>
								<strong>2017-06-27T18:10:05Z</strong>
							</td>
						</tr>
						<tr>
							<td>Processor name</td>
							<td>
								<strong>WP12-DDSS-024 GDM-SAR Sentinel-1 processing on demand</strong>
							</td>
						</tr>
						<tr>
							<td>Orbit direction</td>
							<td>
								<strong>ASCENDING</strong>
							</td>
						</tr>
						<tr>
							<td>Orbit</td>
							<td>
								<strong>001</strong>
							</td>
						</tr>
						<tr>
							<td>Unit of measure</td>
							<td>
								<strong>rad</strong>
							</td>
						</tr>
						<tr>
							<td>Reference system</td>
							<td>
								<strong>RADAR_GEOMETRY</strong>
							</td>
						</tr>
						<tr>
							<td>License</td>
							<td>
								<strong>https://creativecommons.org/licenses/by-sa/4.0/</strong>
							</td>
						</tr>
						<tr>
							<td>Super Master SAR Image</td>
							<td>
								<strong>S1B_IW_SLC__1SDV_20170703T180938_20170703T181005_006327_00B1F9_012B.SAFE</strong>
							</td>
						</tr>
						<tr>
							<td>Master SAR Image</td>
							<td>
								<strong>S1B_IW_SLC__1SDV_20170621T180937_20170621T181004_006152_00ACE9_35A7.SAFE</strong>
							</td>
						</tr>
						<tr>
							<td>Slave</td>
							<td>
								<strong>S1A_IW_SLC__1SDV_20170627T181009_20170627T181036_017223_01CBB8_3A0F.SAFE</strong>
							</td>
						</tr>
						<tr>
							<td>Perpendicular baseline (m)</td>
							<td>
								<strong>-63.2066448083</strong>
							</td>
						</tr>
						<tr>
							<td>DEM</td>
							<td>
								<strong></strong>
							</td>
						</tr>
						<tr>
							<td>Method</td>
							<td>
								<strong>NSBAS processing chain, differential interferogram generation with bursts coregistration using spectral diversity</strong>
							</td>
						</tr>
						<tr>
							<td>Applied filter</td>
							<td>
								<strong></strong>
							</td>
						</tr>
					</tbody>
				</table>
			</content>
			<license
				xmlns="http://www.w3.org/2005/Atom">https://creativecommons.org/licenses/by-sa/4.0/
			</license>
			<producttype
				xmlns="http://www.opengis.net/eop/2.1">WRAPPED_INTERFEROGRAM
			</producttype>
			<georss:polygon
				xmlns:georss="http://www.georss.org/georss">
-4.2197 36.4027 -3.2463 36.5491 -3.5283 37.7806 -4.5184 37.6343 -4.2197 36.4027

			</georss:polygon>
			<box
				xmlns="http://www.georss.org/georss">-4.5184 36.4027 -3.2463 37.7806
			</box>
			<date
				xmlns="http://purl.org/dc/elements/1.1/">
2017-06-21T18:09:22.0000000Z/2017-06-27T18:10:05.0000000Z

			</date>
			<spatial
				xmlns="http://purl.org/dc/terms/">
POLYGON((36.4027 -4.2197,36.5491 -3.2463,37.7806 -3.5283,37.6343 -4.5184,36.4027 -4.2197))

			</spatial>
			<producttype
				xmlns="http://www.opengis.net/eop/2.1">WRAPPED_INTERFEROGRAM
			</producttype>
			<sar:earthobservation
				xmlns:eop="http://www.opengis.net/eop/2.1"
				xmlns:gml="http://www.opengis.net/gml/3.2"
				xmlns:om="http://www.opengis.net/om"
				xmlns:sar="http://www.opengis.net/sar/2.1">
				<phenomenontime
					xmlns="http://www.opengis.net/om/2.0">
					<gml:timeperiod>
						<gml:beginposition>2017-06-21T18:09:22.0000000Z</gml:beginposition>
						<gml:endposition>2017-06-27T18:10:05.0000000Z</gml:endposition>
					</gml:timeperiod>
				</phenomenontime>
				<procedure
					xmlns="http://www.opengis.net/om/2.0">
					<eop:earthobservationequipment>
						<eop:platform>
							<eop:platform>
								<eop:shortname>S1</eop:shortname>
							</eop:platform>
						</eop:platform>
						<eop:instrument>
							<eop:instrument>
								<eop:shortname>SAR</eop:shortname>
								<eop:description>Synthetic Aperture Radar</eop:description>
							</eop:instrument>
						</eop:instrument>
						<eop:sensor>
							<eop:sensor>
								<eop:sensortype>RADAR</eop:sensortype>
								<eop:operationalmode>IW</eop:operationalmode>
							</eop:sensor>
						</eop:sensor>
						<eop:acquisitionparameters>
							<sar:acquisition>
								<eop:orbitnumber>1</eop:orbitnumber>
								<eop:lastorbitnumber/>
								<eop:orbitdirection>ASCENDING</eop:orbitdirection>
								<sar:polarisationchannels>V/V</sar:polarisationchannels>
								<sar:antennalookdirection>RIGHT</sar:antennalookdirection>
							</sar:acquisition>
						</eop:acquisitionparameters>
					</eop:earthobservationequipment>
				</procedure>
				<featureofinterest
					xmlns="http://www.opengis.net/om/2.0">
					<eop:footprint>
						<eop:multiextentof>
							<gml:multisurface>
								<gml:surfacemembers>
									<gml:polygon>
										<gml:exterior>
											<gml:linearring>
												<gml:poslist count="5">
-4.2197 36.4027 -3.2463 36.5491 -3.5283 37.7806 -4.5184 37.6343 -4.2197 36.4027
</gml:poslist>
											</gml:linearring>
										</gml:exterior>
									</gml:polygon>
								</gml:surfacemembers>
							</gml:multisurface>
						</eop:multiextentof>
					</eop:footprint>
				</featureofinterest>
				<result
					xmlns="http://www.opengis.net/om/2.0">
					<eop:earthobservationresult>
						<eop:browse>
							<eop:browseinformation>
								<eop:referencesystemidentifier>RADAR_GEOMETRY</eop:referencesystemidentifier>
								<eop:filename>
									<servicereference
										xmlns="http://www.opengis.net/ows/2.0"
										xmlns:p10="http://www.opengis.net/ows/2.0"
										xmlns:p11="http://www.w3.org/1999/xlink" p10:type="simple" p11:href="https://sandbox.zenodo.org/record/246071/files/GDM_20170621-20170627_sd_era_4rlks.int.png">
										<p10:requestmessage/>
									</servicereference>
								</eop:filename>
							</eop:browseinformation>
						</eop:browse>
						<eop:product>
							<eop:productinformation>
								<eop:referencesystemidentifier>RADAR_GEOMETRY</eop:referencesystemidentifier>
								<eop:filename>
									<servicereference
										xmlns="http://www.opengis.net/ows/2.0"
										xmlns:p10="http://www.opengis.net/ows/2.0"
										xmlns:p11="http://www.w3.org/1999/xlink" p10:type="simple" p11:href="https://sandbox.zenodo.org/record/246071/files/GDM_20170621-20170627_sd_era_4rlks.int.tiff" p11:title="simple">
										<p10:requestmessage/>
									</servicereference>
								</eop:filename>
								<eop:size>4500</eop:size>
							</eop:productinformation>
						</eop:product>
					</eop:earthobservationresult>
				</result>
				<eop:metadataproperty>
					<eop:earthobservationmetadata>
						<eop:identifier>InW_GDM-CNRS_20170621_20170627_0001</eop:identifier>
						<eop:acquisitiontype>OTHER</eop:acquisitiontype>
						<eop:producttype>WRAPPED_INTERFEROGRAM</eop:producttype>
						<eop:status>ARCHIVED</eop:status>
						<eop:statussubtype>ON-LINE</eop:statussubtype>
						<eop:imagequalitystatus>NOMINAL</eop:imagequalitystatus>
						<eop:productqualitystatus>DEGRADED</eop:productqualitystatus>
						<eop:processing>
							<eop:processinginformation>
								<eop:processingcenter>GDM-CNRS</eop:processingcenter>
								<eop:processingdate>2018-10-04T11:03:57Z</eop:processingdate>
								<eop:method>
NSBAS processing chain, differential interferogram generation with bursts coregistration using spectral diversity. Applied filter: . Used DEM:
</eop:method>
								<eop:processorname>
WP12-DDSS-024 GDM-SAR Sentinel-1 processing on demand
</eop:processorname>
								<eop:processorversion>commit_svn:2343 date:2018-03-05 15:29:54</eop:processorversion>
								<eop:processinglevel>EPOS_L1</eop:processinglevel>
								<eop:nativeproductformat>geoTiff</eop:nativeproductformat>
								<eop:auxiliarydatasetfilename/>
							</eop:processinginformation>
						</eop:processing>
						<eop:vendorspecific>
							<eop:specificinformation>
								<eop:localattribute>verticalResolution</eop:localattribute>
								<eop:localvalue>62.4407439728</eop:localvalue>
							</eop:specificinformation>
						</eop:vendorspecific>
					</eop:earthobservationmetadata>
				</eop:metadataproperty>
			</sar:earthobservation>
			<identifier
				xmlns="http://purl.org/dc/elements/1.1/">InW_GDM-CNRS_20170621_20170627_0001
			</identifier>
		</entry>
```

I was trying to collect `inw`. This is the configuration I used:

```json
{
  "collection": "inw",
  "start_date": "2010-01-16T10:30:00Z",
  "timeout": 4,
  "make_private":  false
}
```

There were 6 datasets and all of them failed:

![image](https://user-images.githubusercontent.com/8862002/58549971-07e2ca80-820d-11e9-88c7-1547d108d4c8.png)

davidcordeiro-deimos commented:
Using the same configuration as you presented, I am able to catalog the 9 datasets, as seen in the gather and fetch logs presented bellow. I also attached an screenshot of a finished job with the 9 datasets, as well as an example of a dataset.

The spatial field within the extra field dictionary looks as follows:
{'value': u'{"type": "Polygon", "coordinates": [[[36.4027, -4.2197],[36.5491, -3.2463],[37.7806, -3.5283],[37.6343, -4.5184],[36.4027, -4.2197]]]}', 'key': 'spatial'}

![image](https://user-images.githubusercontent.com/46027889/58549693-fba63f80-8203-11e9-91ed-47b35d7ab427.png)

![image](https://user-images.githubusercontent.com/46027889/58549729-137dc380-8204-11e9-8186-722c2e1230b7.png)

![image](https://user-images.githubusercontent.com/46027889/58549750-22647600-8204-11e9-8e0a-de9e46472c22.png)

![image](https://user-images.githubusercontent.com/46027889/58550636-1974a400-8206-11e9-98c5-48f88d9ad5b9.png)

davidcordeiro-deimos commented:
@georgiana-b Regarding the non-stopping jobs I believe that I solved that problem, and committed the changes into the repository.

However, I think that the error that you were obtaining previously has no connection to this.

Please let me know of any updates

georgiana-b commented:
@davidcordeiro-deimos I just gave it another try after your fix. It does stop now so that's solved. 
However I still get indexing errors like this for multiple files:

```
solr          | 52491 INFO  (qtp1989972246-19) [   x:ckan] o.a.s.c.S.Request [ckan]  webapp=/solr path=/select params={q=name:"inw_gdm-cnrs_20170627_20170703_9ada"+OR+id:"inw_gdm-cnrs_20170627_20170703_9ada"&fq=site_id:"default"&rows=1&wt=json} hits=0 status=0 QTime=2
ckan-dev_1    | 2019-05-29 12:43:43,193 DEBUG [ckanext.spatial.plugin] Received: u'{"type": "Polygon", "coordinates": [[[-4.547082978 36.397081439],[-4.547082978 37.785971439],[-3.217081914 37.785971439],[-3.217081914 36.397081439],[-4.547082978 36.397081439]]]}'
ckan-dev_1    | 2019-05-29 12:43:43,265 ERROR [ckanext.spatial.plugin] Geometry not valid GeoJSON, not indexing
```
And another example

```
ckan-dev_1    | 2019-05-29 12:43:43,793 DEBUG [ckanext.spatial.plugin] Received: u'{"type": "Polygon", "coordinates": [[[36.4027 -4.2197],[36.5491 -3.2463],[37.7806 -3.5283],[37.6343 -4.5184],[36.4027 -4.2197]]]}'
ckan-dev_1    | 2019-05-29 12:43:43,879 ERROR [ckanext.spatial.plugin] Geometry not valid GeoJSON, not indexing
```

georgiana-b commented:
And eventually when the changes are commited to the index I get the initial error again:

```
And eventually when the changes are commited to the index I get the one above:

```
ckan-dev_1    | 2019-05-29 13:07:18,700 ERROR [ckanext.spatial.plugin] Geometry not valid GeoJSON, not indexing
ckan-dev_1    | 2019-05-29 13:07:18,769 ERROR [ckan.lib.search] Error while indexing dataset ddb01785-601e-4534-8d45-cb21a0e0b9e1: ValueError("Expecting ',' delimiter: line 1 column 49 (char 48)",)
ckan-dev_1    | Traceback (most recent call last):
ckan-dev_1    |   File "/usr/bin/paster", line 11, in <module>
ckan-dev_1    |     load_entry_point('PasteScript==2.0.2', 'console_scripts', 'paster')()
ckan-dev_1    |   File "/usr/lib/python2.7/site-packages/paste/script/command.py", line 102, in run
ckan-dev_1    |     invoke(command, command_name, options, args[1:])
ckan-dev_1    |   File "/usr/lib/python2.7/site-packages/paste/script/command.py", line 141, in invoke
ckan-dev_1    |     exit_code = runner.run(args)
ckan-dev_1    |   File "/usr/lib/python2.7/site-packages/paste/script/command.py", line 236, in run
ckan-dev_1    |     result = self.command()
ckan-dev_1    |   File "/srv/app/src/ckan/ckan/lib/cli.py", line 554, in command
ckan-dev_1    |     self.rebuild()
ckan-dev_1    |   File "/srv/app/src/ckan/ckan/lib/cli.py", line 577, in rebuild
ckan-dev_1    |     quiet=self.options.quiet)
ckan-dev_1    |   File "/srv/app/src/ckan/ckan/lib/search/__init__.py", line 194, in rebuild
ckan-dev_1    |     defer_commit
ckan-dev_1    |   File "/srv/app/src/ckan/ckan/lib/search/index.py", line 101, in update_dict
ckan-dev_1    |     self.index_package(pkg_dict, defer_commit)
ckan-dev_1    |   File "/srv/app/src/ckan/ckan/lib/search/index.py", line 278, in index_package
ckan-dev_1    |     pkg_dict = item.before_index(pkg_dict)
ckan-dev_1    |   File "/srv/app/src_extensions/ckanext-nextgeoss/ckanext/nextgeoss/plugin.py", line 231, in before_index
ckan-dev_1    |     geometry = json.loads(geometry)
ckan-dev_1    |   File "/usr/lib/python2.7/json/__init__.py", line 339, in loads
ckan-dev_1    |     return _default_decoder.decode(s)
ckan-dev_1    |   File "/usr/lib/python2.7/json/decoder.py", line 364, in decode
ckan-dev_1    |     obj, end = self.raw_decode(s, idx=_w(s, 0).end())
ckan-dev_1    |   File "/usr/lib/python2.7/json/decoder.py", line 380, in raw_decode
ckan-dev_1    |     obj, end = self.scan_once(s, idx)
ckan-dev_1    | ValueError: Expecting ',' delimiter: line 1 column 49 (char 48)
```

I wonder why you don't have this problem. Are we sure  we are using the same version and schema for solr?
```

davidcordeiro-deimos commented:
> ckan-dev_1    | 2019-05-29 12:43:43,793 DEBUG [ckanext.spatial.plugin] Received: u'{"type": "Polygon", "coordinates": [[[36.4027 -4.2197],[36.5491 -3.2463],[37.7806 -3.5283],[37.6343 -4.5184],[36.4027 -4.2197]]]}'

If you are getting this error, I would say that is due to not having the most recent code in lib/epos_base.py. Can you confirm that in the lines 126 to 128 you have the following:
```
            tmp_list = geojson_tmp.split('))')[0].split('((')[1]
            tmp_list = tmp_list.replace(',', '],[').replace(' ', ', ')
            coords_list = '[[' + tmp_list + ']]'
```
As you can see in the following screenshot, I used the breakpoint to check the value of the 'spatial' field, and it presents ',' (commas) between longitude and latitude. 

![image](https://user-images.githubusercontent.com/46027889/58561461-0b7f4d00-821f-11e9-8e7e-3dd66195cc32.png)

georgiana-b commented:
Yes, I can confirm I have the same code. The value of `geojson` at [that line](https://github.com/NextGeoss/ckanext-nextgeossharvest/blob/2788abb2d9c4607be440092663ef91659d7f5fab/ckanext/nextgeossharvest/lib/epos_base.py#L130) is ok for me as well but what is sent to solr lacks the comma.

davidcordeiro-deimos commented:
@georgiana-b We searched for the dataset in solr, which has commas in the spatial field.

![image](https://user-images.githubusercontent.com/46027889/58563780-a8dc8000-8223-11e9-8c9b-9fcfd9076051.png)

georgiana-b commented:
Hi @davidcordeiro-deimos! I've managed to solve that indexing error and to run the harvester locally once. However, every time I reharvest I get errors for the datasets that already exist (the ones that are being updated).
The errors are not very helpful I'm afraid since all I get is:

```
ckan-dev_1    | 2019-06-06 12:51:36,477 DEBUG [ckanext.harvest.model] Error updating inu_gdm-cnrs_20170627_20170703_0354: None
```

Jovanka run it too and encountered the same issue. Can you please run a harvester multiple times and check what happens?

davidcordeiro-deimos commented:
Hi @georgiana-b, from the error log that you sent, I saw that you are running collection "inu".
I run the harvester multiple times, however I was not able to reproduce that error. Bellow you can find my logs of the gather and fetch stages for the last job. If you notice, in these logs there is no line regarding [ckanext.harvest.model] as in the error that you are obtaining. Also, we can see that no dataset will be updated due to being already collected.

In order to pinpoint the cause of the error that you are obtaining, I would suggest for you to send the complete logs. This way we can see where exactly the error is being raised.

![image](https://user-images.githubusercontent.com/46027889/59258840-a4a36000-8c30-11e9-91e7-7dd1d9995ebc.png)

![image](https://user-images.githubusercontent.com/46027889/59258897-b2f17c00-8c30-11e9-9be1-c9e2a4359f5b.png)

georgiana-b commented:
Hi @davidcordeiro-deimos!

Just ran this on staging to make sure it's not just my local setup. This is the full log of the third run the job:

```
2019-06-11T11:35:37.891135045Z INFO:ckanext.harvest.logic.action.create:Harvest job create: {'source_id': u'd996ec18-b5b7-4435-b348-3d0ad6ad87b7', 'run': True}
2019-06-11T11:35:37.947971381Z INFO:ckanext.harvest.logic.action.create:Harvest job saved 16f11e7f-67d5-4754-9f32-89475ed9c899
2019-06-11T11:35:37.948113622Z INFO:ckanext.harvest.logic.action.update:Send job to gather queue: {'id': u'16f11e7f-67d5-4754-9f32-89475ed9c899'}
2019-06-11T11:35:37.969829199Z INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): solr
2019-06-11T11:35:37.972962986Z INFO:pysolr:Finished 'http://solr:8983/solr/ckan/select/?q=name%3A%22d996ec18-b5b7-4435-b348-3d0ad6ad87b7%22+OR+id%3A%22d996ec18-b5b7-4435-b348-3d0ad6ad87b7%22&fq=site_id%3A%22nextgeoss-staging%22&rows=1&wt=json' (get) with body '' in 0.005 seconds, with status 200
2019-06-11T11:35:38.014469913Z 2019-06-11 11:35:38,012 DEBUG [ckanext.harvest.queue] Received harvest job id: 16f11e7f-67d5-4754-9f32-89475ed9c899
2019-06-11T11:35:38.014592962Z INFO:ckanext.harvest.logic.action.update:Sent job 16f11e7f-67d5-4754-9f32-89475ed9c899 to the gather queue
2019-06-11T11:35:38.029643789Z 2019-06-11 11:35:38,028 DEBUG [ckanext.nextgeossharvest.harvesters.epos.EPOSSat.gather] EPOSSatHarvester gather_stage for job: <HarvestJob id=16f11e7f-67d5-4754-9f32-89475ed9c899 created=2019-06-11 11:35:37.928929 gather_started=2019-06-11 11:35:38.028304 gather_finished=None finished=None source_id=d996ec18-b5b7-4435-b348-3d0ad6ad87b7 status=Running>
2019-06-11T11:35:38.030300627Z 2019-06-11 11:35:38,029 DEBUG [ckanext.nextgeossharvest.lib.nextgeoss_base] Using config: {u'make_private': False, u'timeout': 4, u'start_date': u'2017-01-16T10:30:00Z', u'collection': u'inw'}
2019-06-11T11:35:38.043631133Z 2019-06-11 11:35:38,043 DEBUG [ckanext.nextgeossharvest.harvesters.epos.EPOSSat.gather] Restart page is 1
2019-06-11T11:35:38.044282620Z 2019-06-11 11:35:38,043 DEBUG [ckanext.nextgeossharvest.harvesters.epos.EPOSSat.gather] Harvest URL is https://catalog.terradue.com/gep-epos/search?start=2017-01-16T10:30:00Z&&pt=WRAPPED_INTERFEROGRAM&startIndex=1
2019-06-11T11:35:38.046681152Z INFO:repoze.who:no challenge required
2019-06-11T11:35:38.046733835Z INFO:repoze.who:-- repoze.who request ended (/harvest/refresh/d996ec18-b5b7-4435-b348-3d0ad6ad87b7) --
2019-06-11T11:35:38.108759658Z INFO:repoze.who:request classification: browser
2019-06-11T11:35:38.109664893Z INFO:repoze.who:-- repoze.who request started (/harvest/admin/d996ec18-b5b7-4435-b348-3d0ad6ad87b7) --
2019-06-11T11:35:38.139584898Z INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): solr
2019-06-11T11:35:38.148330921Z INFO:pysolr:Finished 'http://solr:8983/solr/ckan/select/?q=name%3A%22d996ec18-b5b7-4435-b348-3d0ad6ad87b7%22+OR+id%3A%22d996ec18-b5b7-4435-b348-3d0ad6ad87b7%22&fq=site_id%3A%22nextgeoss-staging%22&rows=1&wt=json' (get) with body '' in 0.011 seconds, with status 200
2019-06-11T11:35:38.222696180Z /usr/lib/python2.7/site-packages/requests/packages/urllib3/connectionpool.py:838: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/security.html
2019-06-11T11:35:38.222745759Z   InsecureRequestWarning)
2019-06-11T11:35:38.290084679Z INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): solr
2019-06-11T11:35:38.295533480Z INFO:pysolr:Finished 'http://solr:8983/solr/ckan/select/?sort=score+desc%2C+metadata_modified+desc&fq=%2Bcapacity%3Apublic+harvest_source_id%3A%22d996ec18-b5b7-4435-b348-3d0ad6ad87b7%22+-dataset_type%3Aharvest&fq=%2Bsite_id%3A%22nextgeoss-staging%22&fq=%2Bstate%3Aactive&facet.mincount=1&rows=11&facet=true&q=%2A%3A%2A&facet.limit=50&wt=json&fl=id+validated_data_dict' (get) with body '' in 0.007 seconds, with status 200
2019-06-11T11:35:38.307212001Z INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): solr
2019-06-11T11:35:38.309805432Z INFO:pysolr:Finished 'http://solr:8983/solr/ckan/select/?sort=score+desc%2C+metadata_modified+desc&fq=%2Bcapacity%3Apublic+harvest_source_id%3A%22d996ec18-b5b7-4435-b348-3d0ad6ad87b7%22+-dataset_type%3Aharvest&fq=%2Bsite_id%3A%22nextgeoss-staging%22&fq=%2Bstate%3Aactive&facet.mincount=1&rows=11&facet=true&q=%2A%3A%2A&facet.limit=50&wt=json&fl=id+validated_data_dict' (get) with body '' in 0.004 seconds, with status 200
2019-06-11T11:35:38.328355883Z INFO:repoze.who:no challenge required
2019-06-11T11:35:38.329497761Z INFO:repoze.who:-- repoze.who request ended (/harvest/admin/d996ec18-b5b7-4435-b348-3d0ad6ad87b7) --
2019-06-11T11:35:38.438412929Z INFO:repoze.who:request classification: browser
2019-06-11T11:35:38.438761574Z INFO:repoze.who:-- repoze.who request started (/fanstatic/vendor/:version:2017-09-29T10:01:10/select2/select2.css) --
2019-06-11T11:35:38.442437161Z INFO:repoze.who:request classification: browser
2019-06-11T11:35:38.442476486Z INFO:repoze.who:-- repoze.who request started (/fanstatic/css/:version:2017-09-29T10:01:10/main.min.css) --
2019-06-11T11:35:38.443605817Z INFO:repoze.who:no challenge required
2019-06-11T11:35:38.444027552Z INFO:repoze.who:-- repoze.who request ended (/fanstatic/vendor/:version:2017-09-29T10:01:10/select2/select2.css) --
2019-06-11T11:35:38.446254664Z INFO:repoze.who:no challenge required
2019-06-11T11:35:38.447263492Z INFO:repoze.who:-- repoze.who request ended (/fanstatic/css/:version:2017-09-29T10:01:10/main.min.css) --
2019-06-11T11:35:38.453272610Z INFO:repoze.who:request classification: browser
2019-06-11T11:35:38.454117799Z INFO:repoze.who:-- repoze.who request started (/fanstatic/ckanext-harvest/:version:2019-04-24T20:23:50/styles/harvest.css) --
2019-06-11T11:35:38.456472491Z INFO:repoze.who:request classification: browser
2019-06-11T11:35:38.456490448Z INFO:repoze.who:no challenge required
2019-06-11T11:35:38.456504874Z INFO:repoze.who:-- repoze.who request started (/fanstatic/vendor/:version:2017-09-29T10:01:10/font-awesome/css/font-awesome.min.css) --
2019-06-11T11:35:38.456776720Z INFO:repoze.who:-- repoze.who request ended (/fanstatic/ckanext-harvest/:version:2019-04-24T20:23:50/styles/harvest.css) --
2019-06-11T11:35:38.457468364Z INFO:repoze.who:no challenge required
2019-06-11T11:35:38.457603561Z INFO:repoze.who:-- repoze.who request ended (/fanstatic/vendor/:version:2017-09-29T10:01:10/font-awesome/css/font-awesome.min.css) --
2019-06-11T11:35:38.641398220Z 2019-06-11 11:35:38,640 INFO  [provider_logger] epos         | 2019-06-11 11:35:38.044236 | 200 | 0.457284s
2019-06-11T11:35:38.868031757Z 2019-06-11 11:35:38,867 DEBUG [ckanext.nextgeossharvest.harvesters.epos] inw_cnrirea_20190518_20190530_w452 will not be updated.
2019-06-11T11:35:38.877070032Z /usr/lib/python2.7/site-packages/sqlalchemy/orm/unitofwork.py:79: SAWarning: Usage of the 'related attribute set' operation is not currently supported within the execution stage of the flush process. Results may not be consistent.  Consider using alternative event listeners or connection-level operations instead.
2019-06-11T11:35:38.877152365Z   sess._flush_warning("related attribute set")
2019-06-11T11:35:38.877401036Z /usr/lib/python2.7/site-packages/sqlalchemy/orm/unitofwork.py:37: SAWarning: Usage of the 'collection append' operation is not currently supported within the execution stage of the flush process. Results may not be consistent.  Consider using alternative event listeners or connection-level operations instead.
2019-06-11T11:35:38.877432220Z   sess._flush_warning("collection append")
2019-06-11T11:35:38.890609112Z /usr/lib/python2.7/site-packages/sqlalchemy/orm/session.py:1907: SAWarning: Attribute history events accumulated on 1 previously clean instances within inner-flush event handlers have been reset, and will not result in database updates. Consider using set_committed_value() within inner-flush event handlers to avoid this warning.
2019-06-11T11:35:38.890657969Z   self._flush(objects)
2019-06-11T11:35:38.915754062Z 2019-06-11 11:35:38,914 DEBUG [ckanext.nextgeossharvest.harvesters.epos] inw_fcasu_20181222s1a_20181228s1b_0hvn will not be updated.
2019-06-11T11:35:38.957517054Z 2019-06-11 11:35:38,956 DEBUG [ckanext.nextgeossharvest.harvesters.epos] inw_cnr_irea_20181127s1a_20181203s1b_k1mc will not be updated.
2019-06-11T11:35:38.984574187Z 2019-06-11 11:35:38,983 DEBUG [ckanext.nextgeossharvest.harvesters.epos] inw_cnr_irea_20180502s1a_20180508s1b_xvdv has not been harvested before. Creating a new harvest object.
2019-06-11T11:35:39.012817802Z 2019-06-11 11:35:39,009 DEBUG [ckanext.nextgeossharvest.harvesters.epos] inw_cnr_irea_20170211s1b_20180101s1b_ru11 will not be updated.
2019-06-11T11:35:39.031040512Z INFO:repoze.who:request classification: browser
2019-06-11T11:35:39.031067488Z INFO:repoze.who:-- repoze.who request started (/api/3/action/status_show) --
2019-06-11T11:35:39.031403923Z INFO:repoze.who:no identities found, not authenticating
2019-06-11T11:35:39.040358624Z INFO:repoze.who:no challenge required
2019-06-11T11:35:39.040677242Z INFO:repoze.who:-- repoze.who request ended (/api/3/action/status_show) --
2019-06-11T11:35:39.041776134Z 2019-06-11 11:35:39,040 DEBUG [ckanext.nextgeossharvest.harvesters.epos] inw_gdm_cnrs_20170627_20170703_9ada has not been harvested before. Creating a new harvest object.
2019-06-11T11:35:39.056171951Z INFO:repoze.who:request classification: browser
2019-06-11T11:35:39.056216306Z INFO:repoze.who:-- repoze.who request started (/api/i18n/en) --
2019-06-11T11:35:39.064687616Z 2019-06-11 11:35:39,063 DEBUG [ckanext.nextgeossharvest.harvesters.epos] inw_gdm_cnrs_20170621_20170627_1111 has not been harvested before. Creating a new harvest object.
2019-06-11T11:35:39.081263141Z INFO:repoze.who:no challenge required
2019-06-11T11:35:39.081780179Z INFO:repoze.who:-- repoze.who request ended (/api/i18n/en) --
2019-06-11T11:35:39.090392976Z 2019-06-11 11:35:39,089 DEBUG [ckanext.nextgeossharvest.harvesters.epos] inw_gdm_cnrs_20170621_20170627_0042 has not been harvested before. Creating a new harvest object.
2019-06-11T11:35:39.113845297Z 2019-06-11 11:35:39,113 DEBUG [ckanext.nextgeossharvest.harvesters.epos] inw_gdm_cnrs_20170621_20170627_0002 has not been harvested before. Creating a new harvest object.
2019-06-11T11:35:39.136039401Z 2019-06-11 11:35:39,135 DEBUG [ckanext.nextgeossharvest.harvesters.epos] inw_gdm_cnrs_20170621_20170627_0001 has not been harvested before. Creating a new harvest object.
2019-06-11T11:35:39.149848146Z 2019-06-11 11:35:39,149 INFO  [harvester_logger] epos         | 2019-06-11 11:35:39.149024 | jobID:16f11e7f-67d5-4754-9f32-89475ed9c899 | 6 | 0
2019-06-11T11:35:39.162620890Z 2019-06-11 11:35:39,161 DEBUG [ckanext.harvest.queue] Received from plugin gather_stage: 10 objects (first: [u'88858fb8-b1ef-4a51-b429-4790f9b1f1c1'] last: [u'33db11a6-610f-4dd4-88ac-dd37a6fa1963'])
2019-06-11T11:35:39.165472405Z 2019-06-11 11:35:39,164 INFO  [ckanext.harvest.queue] Received harvest object id: 88858fb8-b1ef-4a51-b429-4790f9b1f1c1
2019-06-11T11:35:39.169168684Z 2019-06-11 11:35:39,168 DEBUG [ckanext.harvest.queue] Sent 10 objects to the fetch queue
2019-06-11T11:35:39.224069693Z 2019-06-11 11:35:39,223 DEBUG [ckanext.nextgeossharvest.lib.nextgeoss_base.import] Import stage for harvest object with GUID 88858fb8-b1ef-4a51-b429-4790f9b1f1c1
2019-06-11T11:35:39.225625541Z 2019-06-11 11:35:39,223 DEBUG [ckanext.nextgeossharvest.lib.nextgeoss_base] Using config: {u'make_private': False, u'timeout': 4, u'start_date': u'2017-01-16T10:30:00Z', u'collection': u'inw'}
2019-06-11T11:35:39.289056015Z 2019-06-11 11:35:39,288 INFO  [ckanext.harvest.queue] Received harvest object id: 8ee44ecb-d0bc-4220-88a1-74fa0c01ef1c
2019-06-11T11:35:39.340846903Z 2019-06-11 11:35:39,339 DEBUG [ckanext.nextgeossharvest.lib.nextgeoss_base.import] Import stage for harvest object with GUID 8ee44ecb-d0bc-4220-88a1-74fa0c01ef1c
2019-06-11T11:35:39.342076153Z 2019-06-11 11:35:39,340 DEBUG [ckanext.nextgeossharvest.lib.nextgeoss_base] Using config: {u'make_private': False, u'timeout': 4, u'start_date': u'2017-01-16T10:30:00Z', u'collection': u'inw'}
2019-06-11T11:35:39.386952424Z 2019-06-11 11:35:39,386 INFO  [ckanext.harvest.queue] Received harvest object id: 6c51657c-d68c-4631-b53c-a2630bc1493a
2019-06-11T11:35:39.508075988Z 2019-06-11 11:35:39,507 DEBUG [ckanext.nextgeossharvest.lib.nextgeoss_base.import] Import stage for harvest object with GUID 6c51657c-d68c-4631-b53c-a2630bc1493a
2019-06-11T11:35:39.509664896Z 2019-06-11 11:35:39,507 DEBUG [ckanext.nextgeossharvest.lib.nextgeoss_base] Using config: {u'make_private': False, u'timeout': 4, u'start_date': u'2017-01-16T10:30:00Z', u'collection': u'inw'}
2019-06-11T11:35:39.565100830Z 2019-06-11 11:35:39,564 INFO  [ckanext.harvest.queue] Received harvest object id: a096aa83-6099-4b55-934f-57a819378603
2019-06-11T11:35:39.765715824Z 2019-06-11 11:35:39,764 DEBUG [ckanext.nextgeossharvest.lib.nextgeoss_base.import] Import stage for harvest object with GUID a096aa83-6099-4b55-934f-57a819378603
2019-06-11T11:35:39.766354727Z 2019-06-11 11:35:39,765 DEBUG [ckanext.nextgeossharvest.lib.nextgeoss_base] Using config: {u'make_private': False, u'timeout': 4, u'start_date': u'2017-01-16T10:30:00Z', u'collection': u'inw'}
2019-06-11T11:35:39.784102128Z 2019-06-11 11:35:39,783 DEBUG [ckanext.nextgeossharvest.lib.nextgeoss_base] Creating new dataset for inw_cnr-irea_20180502s1a_20180508s1b_xvdv
2019-06-11T11:35:39.849788623Z 2019-06-11 11:35:39,849 DEBUG [ckanext.nextgeossharvest.lib.nextgeoss_base] Updating inw_cnr-irea_20180502s1a_20180508s1b_xvdv
2019-06-11T11:35:39.911921682Z /usr/lib/python2.7/site-packages/sqlalchemy/orm/unitofwork.py:79: SAWarning: Usage of the 'related attribute set' operation is not currently supported within the execution stage of the flush process. Results may not be consistent.  Consider using alternative event listeners or connection-level operations instead.
2019-06-11T11:35:39.911988596Z   sess._flush_warning("related attribute set")
2019-06-11T11:35:39.963418643Z 2019-06-11 11:35:39,962 DEBUG [ckanext.spatial.plugin] Received: u'{"type": "Polygon", "coordinates": [[[-156.97414, 18.0745], [-156.97414, 20.187882], [-154.26129, 20.187882], [-154.26129, 18.0745], [-156.97414, 18.0745]]]}'
2019-06-11T11:35:44.252293860Z 2019-06-11 11:35:44,251 DEBUG [ckanext.harvest.model] Error updating inw_cnr-irea_20180502s1a_20180508s1b_xvdv: None
2019-06-11T11:35:44.295500736Z 2019-06-11 11:35:44,294 INFO  [ckanext.harvest.queue] Received harvest object id: 3edea3b0-f3d8-4b52-8a8e-e2a5d391cc5c
2019-06-11T11:35:44.341451464Z 2019-06-11 11:35:44,340 DEBUG [ckanext.nextgeossharvest.lib.nextgeoss_base.import] Import stage for harvest object with GUID 3edea3b0-f3d8-4b52-8a8e-e2a5d391cc5c
2019-06-11T11:35:44.341606032Z 2019-06-11 11:35:44,341 DEBUG [ckanext.nextgeossharvest.lib.nextgeoss_base] Using config: {u'make_private': False, u'timeout': 4, u'start_date': u'2017-01-16T10:30:00Z', u'collection': u'inw'}
2019-06-11T11:35:44.380092986Z 2019-06-11 11:35:44,379 INFO  [ckanext.harvest.queue] Received harvest object id: 44b1bb4b-fe97-4077-9adf-b5203c327b5a
2019-06-11T11:35:44.415885806Z 2019-06-11 11:35:44,415 DEBUG [ckanext.nextgeossharvest.lib.nextgeoss_base.import] Import stage for harvest object with GUID 44b1bb4b-fe97-4077-9adf-b5203c327b5a
2019-06-11T11:35:44.416267072Z 2019-06-11 11:35:44,415 DEBUG [ckanext.nextgeossharvest.lib.nextgeoss_base] Using config: {u'make_private': False, u'timeout': 4, u'start_date': u'2017-01-16T10:30:00Z', u'collection': u'inw'}
2019-06-11T11:35:44.434715935Z 2019-06-11 11:35:44,434 DEBUG [ckanext.nextgeossharvest.lib.nextgeoss_base] Creating new dataset for inw_gdm-cnrs_20170627_20170703_9ada
2019-06-11T11:35:44.474847475Z 2019-06-11 11:35:44,474 DEBUG [ckanext.nextgeossharvest.lib.nextgeoss_base] Updating inw_gdm-cnrs_20170627_20170703_9ada
2019-06-11T11:35:44.551434954Z 2019-06-11 11:35:44,550 DEBUG [ckanext.spatial.plugin] Received: u'{"type": "Polygon", "coordinates": [[[-4.547082978, 36.397081439], [-4.547082978, 37.785971439], [-3.217081914, 37.785971439], [-3.217081914, 36.397081439], [-4.547082978, 36.397081439]]]}'
2019-06-11T11:35:45.013325979Z 2019-06-11 11:35:45,012 DEBUG [ckanext.harvest.model] Error updating inw_gdm-cnrs_20170627_20170703_9ada: None
2019-06-11T11:35:45.041815031Z 2019-06-11 11:35:45,041 INFO  [ckanext.harvest.queue] Received harvest object id: beabbc2b-eb88-4cbf-8588-2f49b60cc3e3
2019-06-11T11:35:45.092828478Z 2019-06-11 11:35:45,091 DEBUG [ckanext.nextgeossharvest.lib.nextgeoss_base.import] Import stage for harvest object with GUID beabbc2b-eb88-4cbf-8588-2f49b60cc3e3
2019-06-11T11:35:45.093795659Z 2019-06-11 11:35:45,092 DEBUG [ckanext.nextgeossharvest.lib.nextgeoss_base] Using config: {u'make_private': False, u'timeout': 4, u'start_date': u'2017-01-16T10:30:00Z', u'collection': u'inw'}
2019-06-11T11:35:45.108762733Z 2019-06-11 11:35:45,108 DEBUG [ckanext.nextgeossharvest.lib.nextgeoss_base] Creating new dataset for inw_gdm-cnrs_20170621_20170627_1111
2019-06-11T11:35:45.144535061Z 2019-06-11 11:35:45,144 DEBUG [ckanext.nextgeossharvest.lib.nextgeoss_base] Updating inw_gdm-cnrs_20170621_20170627_1111
2019-06-11T11:35:45.214762046Z 2019-06-11 11:35:45,214 DEBUG [ckanext.spatial.plugin] Received: u'{"type": "Polygon", "coordinates": [[[36.4027, -4.2197], [36.5491, -3.2463], [37.7806, -3.5283], [37.6343, -4.5184], [36.4027, -4.2197]]]}'
2019-06-11T11:35:45.758309808Z 2019-06-11 11:35:45,757 DEBUG [ckanext.harvest.model] Error updating inw_gdm-cnrs_20170621_20170627_1111: None
2019-06-11T11:35:45.789090469Z 2019-06-11 11:35:45,788 INFO  [ckanext.harvest.queue] Received harvest object id: 4859290b-9cfe-402d-a989-a1071ccc8b67
2019-06-11T11:35:45.838459400Z 2019-06-11 11:35:45,837 DEBUG [ckanext.nextgeossharvest.lib.nextgeoss_base.import] Import stage for harvest object with GUID 4859290b-9cfe-402d-a989-a1071ccc8b67
2019-06-11T11:35:45.838806889Z 2019-06-11 11:35:45,837 DEBUG [ckanext.nextgeossharvest.lib.nextgeoss_base] Using config: {u'make_private': False, u'timeout': 4, u'start_date': u'2017-01-16T10:30:00Z', u'collection': u'inw'}
2019-06-11T11:35:45.853088195Z 2019-06-11 11:35:45,852 DEBUG [ckanext.nextgeossharvest.lib.nextgeoss_base] Creating new dataset for inw_gdm-cnrs_20170621_20170627_0042
2019-06-11T11:35:45.891078806Z 2019-06-11 11:35:45,890 DEBUG [ckanext.nextgeossharvest.lib.nextgeoss_base] Updating inw_gdm-cnrs_20170621_20170627_0042
2019-06-11T11:35:45.972829324Z 2019-06-11 11:35:45,972 DEBUG [ckanext.spatial.plugin] Received: u'{"type": "Polygon", "coordinates": [[[36.4027, -4.2197], [36.5491, -3.2463], [37.7806, -3.5283], [37.6343, -4.5184], [36.4027, -4.2197]]]}'
2019-06-11T11:35:46.365848101Z 2019-06-11 11:35:46,365 DEBUG [ckanext.harvest.model] Error updating inw_gdm-cnrs_20170621_20170627_0042: None
2019-06-11T11:35:46.390701301Z 2019-06-11 11:35:46,390 INFO  [ckanext.harvest.queue] Received harvest object id: aa9ec2e2-894d-4606-a193-82faf9a672f2
2019-06-11T11:35:46.420796439Z 2019-06-11 11:35:46,420 DEBUG [ckanext.nextgeossharvest.lib.nextgeoss_base.import] Import stage for harvest object with GUID aa9ec2e2-894d-4606-a193-82faf9a672f2
2019-06-11T11:35:46.421147873Z 2019-06-11 11:35:46,420 DEBUG [ckanext.nextgeossharvest.lib.nextgeoss_base] Using config: {u'make_private': False, u'timeout': 4, u'start_date': u'2017-01-16T10:30:00Z', u'collection': u'inw'}
2019-06-11T11:35:46.432314425Z 2019-06-11 11:35:46,432 DEBUG [ckanext.nextgeossharvest.lib.nextgeoss_base] Creating new dataset for inw_gdm-cnrs_20170621_20170627_0002
2019-06-11T11:35:46.468686509Z 2019-06-11 11:35:46,468 DEBUG [ckanext.nextgeossharvest.lib.nextgeoss_base] Updating inw_gdm-cnrs_20170621_20170627_0002
2019-06-11T11:35:46.538617632Z 2019-06-11 11:35:46,538 DEBUG [ckanext.spatial.plugin] Received: u'{"type": "Polygon", "coordinates": [[[36.4027, -4.2197], [36.5491, -3.2463], [37.7806, -3.5283], [37.6343, -4.5184], [36.4027, -4.2197]]]}'
2019-06-11T11:35:46.915898426Z 2019-06-11 11:35:46,915 DEBUG [ckanext.harvest.model] Error updating inw_gdm-cnrs_20170621_20170627_0002: None
2019-06-11T11:35:46.941735037Z 2019-06-11 11:35:46,941 INFO  [ckanext.harvest.queue] Received harvest object id: 33db11a6-610f-4dd4-88ac-dd37a6fa1963
2019-06-11T11:35:46.974012880Z 2019-06-11 11:35:46,973 DEBUG [ckanext.nextgeossharvest.lib.nextgeoss_base.import] Import stage for harvest object with GUID 33db11a6-610f-4dd4-88ac-dd37a6fa1963
2019-06-11T11:35:46.974157553Z 2019-06-11 11:35:46,973 DEBUG [ckanext.nextgeossharvest.lib.nextgeoss_base] Using config: {u'make_private': False, u'timeout': 4, u'start_date': u'2017-01-16T10:30:00Z', u'collection': u'inw'}
2019-06-11T11:35:46.986090563Z 2019-06-11 11:35:46,985 DEBUG [ckanext.nextgeossharvest.lib.nextgeoss_base] Creating new dataset for inw_gdm-cnrs_20170621_20170627_0001
2019-06-11T11:35:47.030763971Z 2019-06-11 11:35:47,030 DEBUG [ckanext.nextgeossharvest.lib.nextgeoss_base] Updating inw_gdm-cnrs_20170621_20170627_0001
2019-06-11T11:35:47.120585843Z 2019-06-11 11:35:47,120 DEBUG [ckanext.spatial.plugin] Received: u'{"type": "Polygon", "coordinates": [[[36.4027, -4.2197], [36.5491, -3.2463], [37.7806, -3.5283], [37.6343, -4.5184], [36.4027, -4.2197]]]}'
2019-06-11T11:35:47.553144383Z 2019-06-11 11:35:47,552 DEBUG [ckanext.harvest.model] Error updating inw_gdm-cnrs_20170621_20170627_0001: None
2019-06-11T11:35:49.142577627Z INFO:repoze.who:request classification: browser
2019-06-11T11:35:49.142678669Z INFO:repoze.who:-- repoze.who request started (/api/3/action/status_show) --
2019-06-11T11:35:49.143055062Z INFO:repoze.who:no identities found, not authenticating
2019-06-11T11:35:49.154076030Z INFO:repoze.who:no challenge required
2019-06-11T11:35:49.154137556Z INFO:repoze.who:-- repoze.who request ended (/api/3/action/status_show) --
2019-06-11T11:35:59.255686819Z INFO:repoze.who:request classification: browser
2019-06-11T11:35:59.255737861Z INFO:repoze.who:-- repoze.who request started (/api/3/action/status_show) --
2019-06-11T11:35:59.255748946Z INFO:repoze.who:no identities found, not authenticating
2019-06-11T11:35:59.262841565Z INFO:repoze.who:no challenge required
2019-06-11T11:35:59.262867246Z INFO:repoze.who:-- repoze.who request ended (/api/3/action/status_show) --
2019-06-11T11:36:00.950313280Z crond: USER ckan pid 422 cmd paster --plugin=ckanext-harvest harvester run -c /srv/app/production.ini >> /var/log/cron.log 2>&1
2019-06-11T11:36:03.951016645Z 2019-06-11 11:36:03,354 DEBUG [ckanext.spatial.plugin] Setting up the spatial model
2019-06-11T11:36:03.951109494Z 2019-06-11 11:36:03,433 DEBUG [ckanext.spatial.model.package_extent] Spatial tables defined in memory
2019-06-11T11:36:03.951117224Z 2019-06-11 11:36:03,448 DEBUG [ckanext.spatial.model.package_extent] Spatial tables already exist
2019-06-11T11:36:03.951131247Z 2019-06-11 11:36:03,479 DEBUG [ckanext.harvest.model] Harvest tables defined in memory
2019-06-11T11:36:03.951136060Z 2019-06-11 11:36:03,483 DEBUG [ckanext.harvest.model] Harvest tables already exist
2019-06-11T11:36:03.951148750Z 
2019-06-11T11:36:03.951165262Z 2019-06-11 11:36:03,711 INFO  [ckanext.harvest.logic.action.update] Harvest job run: {}
2019-06-11T11:36:03.951179631Z 2019-06-11 11:36:03,753 INFO  [ckanext.harvest.logic.action.update] Marking job as finished https://catalog.terradue.com/gep-epos/ 16f11e7f-67d5-4754-9f32-89475ed9c899
```

At the end I get these job reports:

![image](https://user-images.githubusercontent.com/8862002/59278256-f3b5b900-8c61-11e9-85fd-495a65f6d09b.png)

If you copy-paste the log into a file to look at it more closely you will see that the dataset ids reported by the epos harvester have a different format than reported in the error. For example:

This is the original id reported by the harvester (only containing "_")
```
 [ckanext.nextgeossharvest.harvesters.epos] inw_gdm_cnrs_20170621_20170627_0042 has not been harvested before. Creating a new harvest object.
```

 Whereas the id from the model there is a `-` between `inw_gdm` and `cnrs_20170621_20170627_0042`

```
[ckanext.nextgeossharvest.lib.nextgeoss_base] Creating new dataset for inw_gdm-cnrs_20170621_20170627_0042
[ckanext.nextgeossharvest.lib.nextgeoss_base] Updating inw_gdm-cnrs_20170621_20170627_0042
[ckanext.spatial.plugin] Received: u'{"type": "Polygon", "coordinates": [[[36.4027, -4.2197], [36.5491, -3.2463], [37.7806, -3.5283], [37.6343, -4.5184], [36.4027, -4.2197]]]}'
[ckanext.harvest.model] Error updating inw_gdm-cnrs_20170621_20170627_0042: None
```

I assume this might have something to do with the error.

georgiana-b commented:
I tried to follow one of the failing datasets locally from the first run to see what happens. So in the first run I get this:

```
[ckanext.nextgeossharvest.harvesters.epos] inu_gdm_cnrs_20170627_20170703_0354 has not been harvested before. Creating a new harvest object.
```
But than the oject created actually has a `-`:

```
ckan-dev_1    | 2019-06-11 11:15:01,561 DEBUG [ckanext.nextgeossharvest.lib.nextgeoss_base] Creating new dataset for inu_gdm-cnrs_20170627_20170703_0354
```

At the and of the first run I actually have the dataset ` inu_gdm-cnrs_20170627_20170703_0354` in my index. But then in the second run of the job I get:

```
[ckanext.nextgeossharvest.harvesters.epos] inu_gdm_cnrs_20170627_20170703_0354 has not been harvested before. Creating a new harvest object.
```
And then the error:

```
ckan-dev_1    | 2019-06-11 14:32:57,241 DEBUG [ckanext.nextgeossharvest.lib.nextgeoss_base] Creating new dataset for inu_gdm-cnrs_20170627_20170703_0354
ckan-dev_1    | 2019-06-11 14:32:57,273 DEBUG [ckanext.nextgeossharvest.lib.nextgeoss_base] Updating inu_gdm-cnrs_20170627_20170703_0354
ckan-dev_1    | 2019-06-11 14:32:57,961 DEBUG [ckanext.harvest.model] Error updating inu_gdm-cnrs_20170627_20170703_0354: None
```

So if it was all going fine it should have known in the second run that the dataset already exists.

Maybe the whole confusion arises from the fact that even in the original sources, sometimes they use a `_` sometimes they use a `-`. 
For example the original id of the dataset that fails above is `InU_GDM-CNRS_20170627_20170703_0354` but there are also datasets with id like `InU_CNR_IREA_20170211S1B_20180101S1B_RU11`.

davidcordeiro-deimos commented:
Thank you for your effort. Indeed it seems that my local repository and the github were out of sync, and that is why I was not obtaining the error on my side.
On github, it seems that there is missing a `.replace('-', '_')`, I will take care of it and retest it on my side. When corrected and commited, I will let you know.

georgiana-b commented:
Hi @davidcordeiro-deimos! Thank you for looking into this.
Today I run the harvester again including your fix. It doesn't  fail anymore on the second run but it doesn't seem to collect anything either.

![image](https://user-images.githubusercontent.com/8862002/59354674-d4805f80-8d25-11e9-82df-66d32c363251.png)

I tried both `inu` and `inw` and got the same result. The second and subsequent runs don't add any new files. Is this normal? Are there no other files at the source?

davidcordeiro-deimos commented:
Hi @georgiana-b Indeed this is the expected behaviour, on the first run collects the datasets, and on the following runs it will only collect datasets that were not previously collected. As I stated in the pull request, currently the provider only has sample datasets (low number). A large quantity of data is expected to start being injected in September of 2019. This is also why I said that the harvester only needed to be run once per collection.

georgiana-b commented:
I see. Thank you for the explanation @davidcordeiro-deimos.

georgiana-b commented:
@JGulic I tested this on staging as well and everything runs as expected. It has my blessing to be merged.
